Add news functions

theo-bnts opened the pull request:
News methods :
    listSets()
    getReplayDownloadLink()
    getWeaponsDetails()
    listWeaponsSpawnChance()
    getGameModeExtendedData()
    listFeaturedCreativeIslands()
    searchIsland()
    listFish()
    getPlayerFishStats()

theo-bnts committed:
Update fortnite-api.js

theo-bnts committed:
Update endpoints.js

theo-bnts committed:
Update fortnite-api.js

theo-bnts committed:
Update fortnite-api.js

theo-bnts committed:
Add files via upload

theo-bnts committed:
Add files via upload

theo-bnts committed:
Add files via upload

benhawley7 commented on lib/endpoints.js:
```diff
@@ -47,5 +48,15 @@ module.exports = {
     listLoot: lang => `${endpoint}/loot/list?lang=${lang}`,
-    getLootDetails: (id, lang) => `${endpoint}/loot/get?id=${id}&lang=${lang}`
+    getLootDetails: (id, lang) => `${endpoint}/loot/get?id=${id}&lang=${lang}`,
+
```
```suggestion
```
nitpick: Whitespace

benhawley7 commented on lib/fortnite-api.js:
```diff
@@ -417,6 +421,127 @@ class FortniteAPI {
+     * @param {String} [options.lang] en, ar, de, es, es-419, fr, it, ja, ko, pl, pt-BR, ru, tr
+     * @returns {Promise}
+     */
+
```
nitpick: whitespace
```suggestion
```

benhawley7 commented on lib/fortnite-api.js:
```diff
@@ -417,6 +421,127 @@ class FortniteAPI {
+     *
+     * @returns {Promise}
+     */
+
```
```suggestion
```

nitpick: whitespace

benhawley7 commented on lib/fortnite-api.js:
```diff
@@ -417,6 +421,127 @@ class FortniteAPI {
+     * @param {String} [options.lang] unknown supported languages
+     * @returns {Promise}
+     */
+
```
```suggestion
```

nitpick: whitespace

benhawley7 commented on package.json:
```diff
@@ -1,34 +1,65 @@
-        "url": "https://github.com/benhawley7/fortnite-api-io.git"
-    },
-    "files": [
-        "index.js",
-        "lib/**"
-    ],
-    "dependencies": {
-        "node-fetch": "^2.6.0"
-    },
-    "devDependencies": {
-        "eslint": "^6.8.0",
-        "nyc": "^15.0.0",
-        "tape": "^4.13.2"
-    }
+    "escapedName": "fortnite-api-io",
+    "rawSpec": "^1.5.2",
+    "saveSpec": null,
+    "fetchSpec": "^1.5.2"
+  },
+  "_requiredBy": [
+    "/"
+  ],
+  "_resolved": "https://registry.npmjs.org/fortnite-api-io/-/fortnite-api-io-1.5.2.tgz",
+  "_shasum": "e2be1d06036dc26b5723d4c529d0a8430397866c",
+  "_spec": "fortnite-api-io@^1.5.2",
+  "_where": "D:\\Documents\\Bots Discord\\Bot CMaker",
```
This stuff shouldn't be here?

benhawley7 commented on lib/fortnite-api.js:
```diff
@@ -417,6 +421,127 @@ class FortniteAPI {
+     * @param {String} [options.lang] en, ar, de, es, es-419, fr, it, ja, ko, pl, pt-BR, ru, tr
+     * @returns {Promise}
+     */
+    getWeaponsDetails(id, options = {}) {
```
I think this should be `getWeaponDetails()` since its one weapon!

benhawley7 commented on lib/fortnite-api.js:
```diff
@@ -417,6 +421,127 @@ class FortniteAPI {
+     * @param {String} mode
+     * @returns {Promise}
+     */
+    listWeaponsSpawnChance(mode) {
```
Likewise, I would say this should be `listWeaponSpawnChances()`!

theo-bnts replied:
I thinking that correct.

benhawley7 replied:
I would say `listWeaponSpawnChances` is the most appropriate name for this method, given the description in the API docs.

theo-bnts replied:
Okay, all is good :)

benhawley7 reviewed (changes requested):
Thanks for the PR @theo-bnts !

Lots of good work for the endpoints!

Just some changes to be made for method names and random whitespace in the file!

There are also some strange properties in the package.json which need removing!

theo-bnts committed:
Update lib/endpoints.js

Co-authored-by: benhawley7 <31001948+benhawley7@users.noreply.github.com>

theo-bnts committed:
Update lib/fortnite-api.js

Co-authored-by: benhawley7 <31001948+benhawley7@users.noreply.github.com>

theo-bnts committed:
Update endpoints.js

theo-bnts committed:
Update fortnite-api.js

theo-bnts committed:
Update lib/fortnite-api.js

Co-authored-by: benhawley7 <31001948+benhawley7@users.noreply.github.com>

theo-bnts committed:
Update lib/fortnite-api.js

Co-authored-by: benhawley7 <31001948+benhawley7@users.noreply.github.com>

theo-bnts committed:
Update package.json

theo-bnts committed:
Update fortnite-api.js

theo-bnts committed:
Update fortnite-api.js

theo-bnts committed:
Update fortnite-api.js

theo-bnts committed:
Update fortnite-api.js

theo-bnts reviewed (commented):
> Thanks for the PR @theo-bnts !
> 
> Lots of good work for the endpoints!
> 
> Just some changes to be made for method names and random whitespace in the file!
> 
> There are also some strange properties in the package.json which need removing!

No problem !

That good, I have patch all :)

theo-bnts committed:
Update fortnite-api.js

theo-bnts committed:
Update fortnite-api.js

benhawley7 reviewed (approved):
Looks good! Thanks for updating the endpoint and also adding the supported languages for each method in the JSDoc.

Before we merge, I think its best we rename `listWeaponsSpawnChance` to `listWeaponSpawnChances` as this reads better to me and is closer to what is written in the API Docs.

A tip for the future is to write more descriptive commit messages. At the moment they are usually something like `Update fortnite-api.js` which doesn't explain what the change to the code is. Some examples could be:

`feature: Added listFIsh method to the client class`
`fix: Added missing language option to the getTournaments method`

(I like to prefix the the message with `fix:` or `feature:` so I know if its a bug fix or not)

theo-bnts committed:
Update fortnite-api.js

theo-bnts committed:
Update endpoints.js

theo-bnts commented:
> Looks good! Thanks for updating the endpoint and also adding the supported languages for each method in the JSDoc.
> 
> Before we merge, I think its best we rename `listWeaponsSpawnChance` to `listWeaponSpawnChances` as this reads better to me and is closer to what is written in the API Docs.
> 
> A tip for the future is to write more descriptive commit messages. At the moment they are usually something like `Update fortnite-api.js` which doesn't explain what the change to the code is. Some examples could be:
> 
> `feature: Added listFIsh method to the client class`
> `fix: Added missing language option to the getTournaments method`
> 
> (I like to prefix the the message with `fix:` or `feature:` so I know if its a bug fix or not)

No problem, I have patched `listWeaponsSpawnChance` to `listWeaponSpawnChances` and I will add more informations on commit messages in the future 👍

benhawley7 committed:
Merge branch 'develop' into master
